### ISSUE TITLE
Extract data transposing/compression from diqube-server

### ISSUE THREAD

**Issue by bgloeckle** [diqube/diqube#53]
We should separate the transposing/compression of data and the import of it. Currently, importing data takes a vast amount of memory, but after the columns have been compressed, a lot of that memory is freed up again. This is not desired in the JVM of the serving servers - they should use their memory for calculating the results of queries. In addition to that, if the server runs out of memory, the JVM process might just get killed by the OS - something we do not want to happen.
Therefore, we should extract the transposing and compression out of the server process into a separate process that prepares input data in such a way that it can be loaded easily by the server. One way this could be done is to produce files which contain a serialized TableShard and which could then be simply de-serialized in the server process. This means that the whole compression etc. is already done by the data-preparation process.
That way there could be a few machines in a cluster which e.g. produce those import-ready files and the server processes simply needs to de-serialize the data then.

In addition to the above, these import-ready files would be helpful when someone wants to update the version of diqube running in a cluster: The new version would need to be distributed and then one could simply stop the old version and start the new version - which should then be able to import the data pretty quickly.

We should think about providing the logic to create those import-ready files as hadoop output formats, too, to be somewhat compatible to the hadoop ecosystem.

**Comment by bgloeckle** [comment]
When designing the data format of the import-ready files we might weant to keep in mind that we later might want to not hold all the data of the tables in memory any more, but evict some of the rarely-used data on HDD.

**Comment by bgloeckle** [comment]
Implemented de-/serialization of objects and added tests for de-/serialization itself. Created new executable jar diqube-transpose which loads a JSON or CSV file and creates a .diqube file (loads data, compresses and serializes it and writes it to a file). Those .diqube files can now be loaded in the standard way using a .control file with data type "diqube".

**Comment by bgloeckle** [comment]
More extensive tests to be written after #38 is done.
